### DISCUSSION TITLE
How to create a "Node" that behaves like a "Mark"?

### DISCUSSION THREAD

**Discussion by cusxio** [ueberdosis/tiptap#2293] [General Questions]
## Desire

I would like to extend the `@tiptap/extension-link` with `addNodeView` to render a NextJs `link` component.

But unfortunately, that's not possible as described below.

## Current Behaviour

`Mark` does not support `addNodeView` as stated in https://github.com/ueberdosis/tiptap/issues/1152#issuecomment-769705282. So my plan is to re-create the `@tiptap/extension-link` via `Node`, so that I can use `addNodeView`.

## Problem

Below is a piece of code I'm currently working on, what do I need to do in the `handler` function to make the text matched into a `Mark`? Any pointers is most welcome!

PS:

I tried using `addMark` but am not sure what should the third parameter be for the function. 

```tsx
export const InternalLink = Node.create({
  name: "internal-link",
  group: "inline",
  inline: true,
  atom: false,
  addNodeView() {
    return ReactNodeViewRenderer(NextJsLink);
  },

  addInputRules() {
    return [
      new InputRule({
        find: (text) => {
          const [match] = linkify.match(text) || [null];

          return match ? { index: match.index, text: match.text } : null;
        },
        handler: ({ state, range, match }) => {
          // What should I do here?
        },
      }),
    ];
  },
});

```

**Comment by wengtytt** [comment]
Any update about this? I have the exactly same request!
